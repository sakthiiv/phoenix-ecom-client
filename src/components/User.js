import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import categoryApi from "../api/category";
import Category from "./Category";
import sessionstorage from '../helpers/sessionstorage';

class CategoryList extends React.Component {
  constructor() {
    super();
    this.state = {
      categories: [],
      redirectToReferrer:false
    };
  }

  async componentDidMount() {
    if(sessionstorage.getItemFromSession('user') && sessionstorage.getItemFromSession('user').role === 'admin') {
      this.setState({redirectToReferrer:true});
    } else {
      this.setState({redirectToReferrer:false});
      const categories = await categoryApi.getAll();
      this.setState({ categories });
    }
  }

  render() {
    console.log(this.state.categories);
    const location = {
      pathname: '/admin',
      state: { fromDashboard: true }
    }
    if (this.state.redirectToReferrer) {
      // console.log('redirecting to admin')
      return <Redirect to={location} />;
    }
    return (
      <div>
        {this.state.categories.length === 0 ? <h1>Something went wrong</h1> : ''}
        <div className="plist_container">
          {this.state.categories.map((category, i) => {
            return <div className="col-sm-3" key={category.id}><Category  category={category} /></div>;
          })}
        </div>
      </div>
    );
  }
}

export default CategoryList;
