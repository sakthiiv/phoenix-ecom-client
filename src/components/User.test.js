import React from "react";
import { shallow } from "enzyme";
import User from "./User";

describe("User", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<User />);
    expect(wrapper).toMatchSnapshot();
  });
});
