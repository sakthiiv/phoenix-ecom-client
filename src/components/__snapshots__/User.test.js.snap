// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders correctly 1`] = `
<div>
  <Category
    category={
      Object {
        "description": "Electronics descriptions goes here.",
        "id": "11",
        "name": "Electronics",
      }
    }
    key="11"
  />
  <Category
    category={
      Object {
        "description": "Clothes descriptions goes here.",
        "id": "22",
        "name": "Clothes",
      }
    }
    key="22"
  />
</div>
`;
